### PR TITLE
Support onednn gemm pre-packing on X86

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/xla/ral/context/common_context_impl_mkldnn.cc [hunk 1]
```diff
@@ -97,6 +97,22 @@ bool enableInOutLayoutTuning() {
   return enabled;
 }
 
+bool initEnableWeightPrePacking() {
+  const char* env = getenv("DISC_CPU_ENABLE_WEIGHT_PRE_PACKING");
```
[the remark this answers is not in the document]
Comment (wyzero) [reply tao_compiler/mlir/xla/ral/context/common_context_impl_mkldnn.cc:101]: Done!

File: tao_compiler/mlir/xla/ral/context/mkldnn/ideep/ideep/operators/matmul.hpp [hunk 2]
```diff
@@ -7,13 +7,13 @@ struct matmul_forward : public dnnl::matmul,
                         utils::computation_cache<dnnl::matmul::primitive_desc> {
   using super = dnnl::matmul;
 
-  template <bool keep_format = true>
+  template <bool keep_format = true, bool weight_format_any = false>
```
[the remark this answers is not in the document]
Comment (wyzero) [reply tao_compiler/mlir/xla/ral/context/mkldnn/ideep/ideep/operators/matmul.hpp:10]: `ideep` is a very old version of onednn, and is mainly only used for pytorch currently. `ideep` itself is just a thin wrapper over onednn, making it more easier to use. `ideep` can benefit from newer onednn version immediately and
we do not upgrade the ideep part as frequent as onednn part.

[the remark this answers is not in the document]
Comment (minminsun) [reply tao_compiler/mlir/xla/ral/context/mkldnn/ideep/ideep/operators/matmul.hpp:10]: I see, thanks.

**Review by minminsun — APPROVED** [review]
LGTM.

**Pull request merged**